src/pygeom.c

```diff
@@ -199,16 +199,26 @@ static PyObject* GeometryObject_ToWKB(GeometryObject* obj) {
 
 static PyObject* GeometryObject_repr(GeometryObject* self) {
   PyObject *result, *wkt, *truncated;
+  char geom_type;
 
   wkt = GeometryObject_ToWKT(self);
   // we never want a repr() to fail; that can be very confusing
   if (wkt == NULL) {
     PyErr_Clear();
     return PyUnicode_FromString("<shapely.Geometry Exception in WKT writer>");
   }
+
+  GEOS_INIT;
+  GEOSGeometry* geom = self->ptr;
+  geom_type = GEOSGeomTypeId_r(ctx, geom);
+  GEOS_FINISH;
+
+  int type_lengths[8] = {5, 10, 10, 7, 10, 15, 12, 18};
+  int max_length = 80 - (11 + type_lengths[geom_type]);
```

Comment:
@caspervdw I was thinking that it would be nice that all reprs have the exact same max length (eg when you have an array with a mixture of geometry types, that the repr truncates at the same place). 
Now, while the above does that, we can probably just as well do this in Python (which would make this simpler)? I don't think there is a specific benefit of keeping the repr definition in C? (the full (non-truncated) WKT string python object is constructed anyway)

(and to be clear, if we keep it in C, the above needs some cleanup to properly handle possible errors)

Reply:
I agree on moving the  repr logic to Python space. Am I right that the Geometry (base)class is now internal (not user-facing)? If not maybe good to keep the original repr (in C) around, else, I am in favour of dropping that altogether.

Reply:
> Am I right that the Geometry (base)class is now internal (not user-facing)? If not maybe good to keep the original repr (in C) around, else, I am in favour of dropping that altogether.

At the moment I didn't yet remove the public exposure, but that's indeed the idea. Although I might actually prefer to keep the class exposed as the "base class" (eg to use for isinstance check), something to further discuss in https://github.com/shapely/shapely/issues/1330